### PR TITLE
Running all the tests on two clusters and only full-rr on 3 clusters

### PR DESCRIPTION
**Pull request by jkremser**
Making `TestFullRoundRobin` dynamic in a way that it can run the assertions properly for any number of clusters (any ~ [2,8])

In makefile we should pass the build tag `all` xor `rr_multicluster` based on the number of clusters that currently runs. `all` will run the all the tests (same as before). `rr_multicluster` will run only the TestFullRoundRobin that depends on `CLUSTERS_NUMBER` env variable (which defaults to `2`).

This way, we can also have a on dispatch GH action that can run the tests on-demand w/ the number of clusters as an input parameter of the action (not done in this PR).

Some data:
I've tried 4 attempts for these pipelines:
- https://github.com/k8gb-io/k8gb/actions/runs/1495842167
- https://github.com/k8gb-io/k8gb/actions/runs/1495842172
- https://github.com/k8gb-io/k8gb/actions/runs/1495842169

All 12 passed so I am assuming it's better than current master where it's pretty fragile.

Signed-off-by: Jirka Kremser <jiri.kremser@gmail.com>

### CODE REVIEW

File: terratest/test/k8gb_abstract_full_roundrobin_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	for _, f := range from {
+		if _, found := aux[f]; !found {
+			diff = append(diff, f)
+		}
+	}
+	return
+}
+
+// helper function for de-duplicating elements in slice (in O(n))
+func dedupe(s []string) (uniq []string) {
```
Comment (kuritka) [inline terratest/test/k8gb_abstract_full_roundrobin_test.go:111]: please rename the function to something more meaningful  (e.g. in SQL wa have distinct keyword for that)

File: terratest/test/k8gb_abstract_full_roundrobin_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"testing"
+)
+
+func AbstractTestFullRoundRobin(t *testing.T, n int) {
+	if n < 2 || n > 8 {
+		t.Logf("Use value of n that represents the number of clusters from interval [2,8]")
+		t.FailNow()
+	} else {
+		t.Logf(fmt.Sprintf("Running TestFullRoundRobin for %d clusters", n))
+	}
```
Comment (kuritka) [inline terratest/test/k8gb_abstract_full_roundrobin_test.go:33]: avoid if else statements if you can, just invert condition and fail fast (e.g: [crossplane doc](https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#return-early)).

**Review by kuritka — CHANGES_REQUESTED** [review]
`.github/workflows/terratest-more-clusters.yaml` will be executed simultaneousy with terratest.yaml workflow ?

File: terratest/test/k8gb_abstract_full_roundrobin_test.go (hunk 2 above)
Comment (jkremser) [reply terratest/test/k8gb_abstract_full_roundrobin_test.go:33]: right, the else branch is totally useless here :D (done)

File: terratest/test/k8gb_abstract_full_roundrobin_test.go (hunk 1 above)
Comment (jkremser) [reply terratest/test/k8gb_abstract_full_roundrobin_test.go:111]: changed to `distinct`

**Comment by jkremser** [comment]
> .github/workflows/terratest-more-clusters.yaml will be executed simultaneousy with terratest.yaml workflow ?

y, that's how it's done in this PR. But I am open to also run it either on demand by dispatch action or nightly/"morningly" (cron), but as it was mentioned on the call, this way nobody will pay attention to the result and it may be hard to track the change that introduced some issue. Btw. it runs only `TestFullRoundRobin` nothing else.

File: terratest/test/k8gb_abstract_full_roundrobin_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+package test
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"k8gbterratest/utils"
+	"testing"
+)
+
+func AbstractTestFullRoundRobin(t *testing.T, n int) {
```
Comment (ytsarev) [inline terratest/test/k8gb_abstract_full_roundrobin_test.go:27]: Do we need to export this function? Isn't it the same test package where it is accessed?

Comment (jkremser) [reply terratest/test/k8gb_abstract_full_roundrobin_test.go]: right, changed to lower case

File: terratest/test/k8gb_lifecycle_test.go [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// +build basic all
```
[the remark this answers is not in the document]
Comment (jkremser) [reply terratest/test/k8gb_lifecycle_test.go]: amended

File: .licignore [hunk 5]
```diff
@@ -11,3 +11,6 @@
 /api/**
 main.go
 gslb_controller.go
+
+# Build tags in tests...
+/terratest/test/**
```
[the remark this answers is not in the document]
Comment (jkremser) [reply .licignore:16]: hmm, yes, but currently it's 8/10 files that have a build tag on them in that dir

[the remark this answers is not in the document]
Comment (ytsarev) [reply .licignore:16]: ok, i guess we can live with that

**Review by ytsarev — APPROVED** [review]
LGTM, thanks for the fixes!

File: .licignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (kuritka) [reply .licignore:16]: I created [issue for that](https://github.com/AbsaOSS/golic/issues/4). Later we can set it up exactly as we need.

**Review by kuritka — APPROVED** [review]

**Pull request merged**